### PR TITLE
pkg/alertmanager: add support for `api_key_file` in OpsGenie config

### LINKED ISSUE #4714: opsgenie_configs field api_key_file not found in type config.plain
**What happened?**

**prometheus-operator** not able to sync **alertmanager** configuration

**Environment**

* Prometheus Operator version:

    `image: quay.io/prometheus-operator/prometheus-operator:v0.55.0`

* Alertmanager version:

    `image: quay.io/prometheus/alertmanager:v0.24.0`

* Alertmanager config:

```
        global:
          resolve_timeout: 5m
        route:
          group_by: ['job']
          group_wait: 10s
          group_interval: 5m
          repeat_interval: 3h
          receiver: 'opsgenie'
          routes:
          - matchers:
            - alertname = Watchdog
            receiver: 'null'
        receivers:
        - name: 'null'
        - name: 'opsgenie'
          opsgenie_configs:
          - send_resolved: true
            api_key_file: /mnt/secrets-store/opsgenie-api-key
```

* Prometheus Operator Logs:

```
{"caller":"klog.go:116","component":"k8s_client_runtime","func":"ErrorDepth","level":"error","msg":"Sync \"monitoring/prometheus-alertmanager\" failed: provision alertmanager configuration: base config from Secret could not be parsed: yaml: unmarshal errors:\n  line 11: field api_key_file not found in type config.plain","ts":"2022-04-13T06:52:45.390938478Z"}
```

### PR DESCRIPTION
**Pull request by simonpasquier**
## Description

Fixes #4714

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [X] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
Fixed support for OpsGenie configurations with `api_key_file`. 
```

### CODE REVIEW

File: pkg/alertmanager/amcfg.go [hunk 1]
```diff
@@ -1382,19 +1382,40 @@ func (r *receiver) sanitize(amVersion semver.Version, logger log.Logger) error {
...
+		level.Warn(logger).Log("msg", "'api_key' and 'api_key_file' are mutually exclusive for OpsGenie receiver config - 'api_key' has taken precedence")
+		ogc.APIKeyFile = ""
+	}
+
+	if ogc.APIKeyFile == "" {
+		return nil
+	}
+
+	if lessThanV0_24 {
+		msg := "'api_key_file' supported in AlertManager >= 0.22.0 only - dropping field from provided config"
```
Comment (slashpai) [inline pkg/alertmanager/amcfg.go:1413]: This should be `>= 0.24.0` in msg ?

Comment (simonpasquier) [reply pkg/alertmanager/amcfg.go]: :+1:

**Comment by slashpai** [comment]
lgtm

**Review by slashpai — APPROVED** [review]

**Review by PhilipGough — APPROVED** [review]
lgtm

**Pull request merged**